dev_guide/getting_traffic_into_cluster.adoc

```diff
@@ -133,10 +146,212 @@ xref:../admin_guide/high_availability.adoc#admin-guide-high-availability[VIP] ad
+situation occurs, the service which requested the port first is given use of the
+port.
+
+xref:using-ingress-IP-self-service[Ingress IP Self-Service] provides a solution.
+
+[[using-ingress-IP-self-service]]
+=== Using Ingress IP Self-Service
+
+Ingress IP Self-Service streamlines the allocation of External IPs for accessing
+services in the cluster. Cluster administrators can designate a range of
+addresses using a CIDR notation, which allows an application user to make a
+request against the cluster for an external IP address. When a service is
+configured with the type `LoadBalancer`, an External IP address will be
+automatically assigned from the designated range.
+
+[NOTE]
+====
+Ingress IP Self-Service is only applicable in non-cloud based environments.
+====
+
+A common use case for Ingress IP Self-Service  is the ability to provide
+database services, such as PostgreSQL, to clients outside of the {product-title}
+cluster.
+
+[[using-ingress-defining-the-ingress-IP-range]]
+==== Defining the Ingress IP Range
+
+The ability for cluster administrators to automatically allocate External IP
+addresses using Ingress is enabled by default within {product-title} and
+configured to use the `172.46.0.0/16` range. An alternate range can be specified
+by configuring the `ingressIPNetworkCIDR` parameter in the
+*_/etc/origin/master-config.yaml_* file:
+
+----
+networkConfig:
+  ingressIPNetworkCIDR: 10.9.54.0/25
+----
+
+Restart the {product-title}  master service to apply the changes.
+
+[[using-ingress-deploy-a-sample-application]]
+==== Deploy a Sample Application
+
+To expose a PostgreSQL as a service for external consumption, the application
+must be first deployed into the cluster. Create a new project or use and
+existing project and instantiate one of the PostgreSQL
+xref:../dev_guide/templates.adoc#dev-guide-templates[templates].
+
+[CAUTION]
+====
+The *postgresql-ephemeral* template does not make use of persistent storage. Once
+the application is scaled down or destroyed, any existing data will be lost. To
+use persistent storage, specify the *postgresql-persistent* template instead.
+====
+
+After instantiating the template, a ClusterIP-based service and
+`DeploymentConfig` is created and a new pod containing PostgreSQL will be
+started.
+
+[[configuring-an-IP-address-for-a-service]]
+==== Configuring an IP Address for a Service
+
+To allow the cluster to automatically assign an IP address for a service, create
+a service definition similar to the following that will create a new Ingress
+service:
+
+----
+$ oc create -f - <<INGRESS
+apiVersion: v1
+kind: Service
+metadata:
+  name: postgresql-ingress
+spec:
+  ports:
+  - name: postgresql
+    port: 5432
+  type: LoadBalancer <1>
+  selector:
+    name: postgresql
+INGRESS
+----
+<1> The `LoadBalancer` type of service will make the request for an external service
+on behalf of the application user.
+
+Alternatively, the `oc expose` command can be used to create the service:
```

Comment:
No, it can do both.  Surprise!

If you expose a dc, rc, or pod you get a service.

If you expose a service, you get a route.